lib/pos_blockchain.go

```diff
@@ -374,17 +366,51 @@ func (bc *Blockchain) getLineageFromCommittedTip(desoBlock *MsgDeSoBlock) ([]*Bl
 	return ancestors, nil
 }
 
-// addBlockToBlockIndex adds the block to the block index and uncommitted blocks map.
+// addBlockToBlockIndex adds the block to the block index with the given status. It also writes the block to the block
+// index in badger.
 func (bc *Blockchain) addBlockToBlockIndex(desoBlock *MsgDeSoBlock, blockStatus BlockStatus) error {
```

Comment:
Now that each block can be added to the index and then have its status updated, should this function act as an upsert instead? I.e. if the block doesn't exist in the index, then this function adds it; otherwise, this function updates the existing node

Reply:
This happens in a branch upstream.